geodesy/haversine_distance.py

```diff
@@ -0,0 +1,48 @@
+from math import asin, atan, cos, sin, sqrt, tan, pow, radians
+
+
+def haversine_distance(lat1, lon1, lat2, lon2):
+    """
+    Calculate great circle distance between two points in a sphere,
+    given longitudes and latitudes. 
+    (https://en.wikipedia.org/wiki/Haversine_formula)
+
+    Args:
+        lat1, lon1: latitude and longitude of coordinate 1
+        lat2, lon2: latitude and longitude of coordinate 2
+        
+    Returns: 
+        geographical distance between two points in metres
+
+    >>> int(haversine_distance(37.774856, -122.424227, 37.864742, -119.537521)) # From SF to Yosemite
+    254352
```

Comment:
What is the unit of measure in great circle calculations?  Google maps says 164 miles or 264 kilometers but that is road distance, not as-the-crow-flies distance.  Helping readers connect what they are learning with what they already know will improve the value of this submission.

Reply:
I'm planning on adding two more algorithms which improve approximation so haversine is a good starter method. Perhaps, a readme file can follow with comparisons and detailed explanations. For now:

We know that the globe is "sort of" spherical, so a path between two points 
    isn't exactly a straight line. We need to account for the Earth's curvature 
    when calculating distance from point A to B. This effect is negligible for 
    small distances but adds up as distance increases. The Haversine method treats
    the earth as a sphere which allows us to "project" the two points A and B 
    onto the surface of that sphere and approximate the spherical distance between
    them. Since the Earth is not a perfect sphere, other methods which model the
    Earth's ellipsoidal nature are more accurate but a quick and modifiable
    computation like Haversine can be handy for shorter range distances.

Reply:
What is the unit of measure?  Is it meters, kilometres, miles, lightyears?

Reply:
Line 15 - metres.. Clarified that in the latest commit as well.

 unit of measure meant is irrelevant for explanation.